render/allocator/allocator.c

```diff
@@ -17,6 +20,26 @@ void wlr_allocator_init(struct wlr_allocator *alloc,
 	wl_signal_init(&alloc->events.destroy);
 }
 
+/* Re-open the DRM node to avoid GEM handle ref'counting issues. See:
+ * https://gitlab.freedesktop.org/mesa/drm/-/merge_requests/110
+ * TODO: don't assume we have the permission to just open the DRM node,
+ * find another way to re-open it.
```

Comment:
Does `dup(2)` work or do we have to reopen by name?

Reply:
Unfortunately dup(2) doesn't work here, because the GEM handlers are per file description. dup(2) only gives a new file descriptor, but shares the same file description on the kernel side. See the linked libdrm discussion above.